test/test_fat.rb

```diff
@@ -5,6 +5,40 @@
 class FatFileTest < Minitest::Test
   include Helpers
 
+  def test_empty_file
+    empty_file = tempfile_with_data("empty_file", "")
+
+    assert_raises MachO::NotAMachOError do
+      MachO::FatFile.new(empty_file)
+    end
+  ensure
+    empty_file.close
+    empty_file.unlink
```

Comment:
Yes, turning `tempfile_with_data` into a method that accepts a block that performs the actual test that involves the temporary file also sounds like a good solution.